*** "..\\libopenjtalk\\jpcommon\\jpcommon_label.c"	Wed Mar  9 18:25:08 2016
--- jpcommon_label.c	Wed Mar  9 18:43:27 2016
***************
*** 296,301 ****
--- 296,302 ----
        if (index == a)
           break;
     }
+    if (i > 3) i = 3;
     return i;
  }
  
***************
*** 395,400 ****
--- 396,402 ----
  
     for (i = 0, index = m->next; index != NULL; index = index->next)
        i++;
+    if (i > 10) i = 10;
     return index_mora_in_utterance(m) + i;
  }
  
***************
*** 429,436 ****
                                           label->phoneme_tail, NULL, NULL);
           label->phoneme_tail = label->phoneme_tail->next;
        } else {
!          fprintf(stderr,
                   "WARNING: JPCommonLabel_insert_word() in jpcommon_label.c: First mora should not be short pause.\n");
        }
        label->short_pause_flag = 0;
     }
--- 431,440 ----
                                           label->phoneme_tail, NULL, NULL);
           label->phoneme_tail = label->phoneme_tail->next;
        } else {
! #if 0
! 		  fprintf(stderr,
                   "WARNING: JPCommonLabel_insert_word() in jpcommon_label.c: First mora should not be short pause.\n");
+ #endif
        }
        label->short_pause_flag = 0;
     }
